base/server/python/pki/server/deployment/__init__.py

```diff
@@ -804,20 +804,29 @@ def pki_connect(self, subsystem):
 
     def import_cert_request(self, subsystem, tag, request):
 
-        logger.info('Creating request ID for %s cert', tag)
-        request.systemCert.requestID = self.client.createRequestID(request)
-        logger.info('- request ID: %s', request.systemCert.requestID)
+        request_id_generator = subsystem.config.get('dbs.request.id.generator', 'legacy')
+
+        if request_id_generator == 'legacy':
+            # get a new request ID from the server
+            logger.info('Creating request ID for %s cert', tag)
+            request.systemCert.requestID = self.client.createRequestID(request)
+            logger.info('- request ID: %s', request.systemCert.requestID)
```

Comment:
This and (new) line 828 are identical and could be factored out, but may be less readable if you do - up to you.

Reply:
They actually serve different purposes. The one on line 813 will show the legacy ID generated by the server before importing the cert request, and the one on line 828 will show the RSNv3 ID generated locally. If we remove 828 we won't see the RSNv3 ID, and if we remove 813 we won't see the legacy ID until the cert request is imported, so if the import fails we won't see the legacy ID at all, which might be useful for troubleshooting.

Reply:
Ah yes, fair enough.